infra/gcp/ensure-conformance-storage.sh

```diff
@@ -0,0 +1,100 @@
+#!/usr/bin/env bash
+#
+# Copyright 2019 The Kubernetes Authors.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+# This script is used to create a new "conformance" bucket in GCS.
+#
+# Each conformance bucket exists in its own GCP project, and is writable by a
+# dedicated googlegroup.
+
+set -o errexit
+set -o nounset
+set -o pipefail
+
+SCRIPT_DIR=$(dirname "${BASH_SOURCE[0]}")
+. "${SCRIPT_DIR}/lib.sh"
+
+function usage() {
+    echo "usage: $0 [repo...]" > /dev/stderr
+    echo "example:" > /dev/stderr
+    echo "  $0 # do all staging repos" > /dev/stderr
+    echo "  $0 coredns # just do one" > /dev/stderr
+    echo > /dev/stderr
+}
+
+# NB: Please keep this sorted.
+CONFORMANCE_PROJECTS=(
+    capi-openstack
+)
+if [ $# = 0 ]; then
+    # default to all staging projects
+    set -- "${CONFORMANCE_PROJECTS[@]}"
+fi
+
+for REPO; do
+    color 3 "Configuring staging: ${REPO}"
+
+    # The GCP project name.
+    PROJECT="k8s-conformance-${REPO}"
+
+    # The group that can write to this staging repo.
+    WRITERS="k8s-infra-conformance-${REPO}@kubernetes.io"
+
+    # The names of the buckets
+    STAGING_BUCKET="gs://${PROJECT}" # used by humans
+    ALL_BUCKETS=("${STAGING_BUCKET}")
+
+    # A short expiration - it can always be raised, but it is hard to lower
+    # We expect promotion within 60d, or for testing to "move on", but
+    # it is also short enough that people should notice occasionally,
+    # and not accidentally think of the conformance buckets as permanent.
+    AUTO_DELETION_DAYS=60
+
+    # Make the project, if needed
+    color 6 "Ensuring project exists: ${PROJECT}"
+    ensure_project "${PROJECT}"
+
+    # Enable writers to use the UI
+    color 6 "Empowering ${WRITERS} as project viewers"
+    empower_group_as_viewer "${PROJECT}" "${WRITERS}"
+
+    # Every project gets some GCS buckets
+
+    # Enable GCS APIs
+    color 6 "Enabling the GCS API"
```

Comment:
I assume you'll want to `ensure_gcs_bucket_retention` too ?

Reply:
I added 10y retention. Did I get it correctly: 
* no files are deleted automatically
* only files older than 10y can be deleted manually?

Reply:
correct